### PR TITLE
48013_create_course_by_supervisor

### PR DESCRIPTION
**Pull request by zenith19**

### CODE REVIEW

**Comment by rofiuddin** [comment]
ok.

**Comment by abdullahallmehedi** [comment]
ok.

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,6 @@
+continue
```
Comment (chuanhtuan) [inline .byebug_history]: move to .gitignore

File: app/controllers/supervisor/courses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def show
+    @course = Course.find params[:id]
+  end
+
+  private
+  def course_params
+    params.require(:course).permit :name, :description, :start_date, :end_date,
+      courses_subjects_attributes: [:id, :course_id, :subject_id]
```
Comment (chuanhtuan) [inline app/controllers/supervisor/courses_controller.rb]: I want, you do not use nested attribute for this feature :D
please google about many-to-many in rails

File: app/controllers/supervisor/courses_controller.rb [hunk 3]
```diff
@@ -6,7 +6,27 @@ def index
...
+      flash[:notice] = t 'flash_course_created'
+      redirect_to [:supervisor, @course]
+    else
+      render :new
+    end
+  end
+
+  private
+  def course_params
+    params.require(:course).permit :name, :descripton, :start_date, :end_date
```
Comment (chuanhtuan) [inline app/controllers/supervisor/courses_controller.rb]: params.require(:course).permit :name, :descripton, :start_date, :end_date, 'subject_ids' => []

File: app/controllers/supervisor/courses_controller.rb [hunk 4]
```diff
@@ -6,7 +6,28 @@ def index
...
+
   def show
     @course = Course.find params[:id]
   end
-end
\ No newline at end of file
+
+  def create
+    @course = Course.new course_params
+    @course.attributes = {'subject_ids' => []}.merge(params[:course] || {})
```
Comment (chuanhtuan) [inline app/controllers/supervisor/courses_controller.rb]: remove

File: app/views/supervisor/courses/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+      <div class="form-group">
+        <%= f.label :end_date %>
+        <%= f.date_field :end_date, class: 'datepicker', id: 'ed', readonly: true %>
+      </div>
+      
+      <div class="form-group">
+        <% Subject.all.each do |subject| -%>
+          <%= check_box_tag :subject_ids, subject.id, @course.subjects.include?(subject),
+            :name => 'course[subject_ids][]'-%>
```
Comment (chuanhtuan) [inline app/views/supervisor/courses/new.html.erb]: name: 'course[subject_ids][]'

File: app/views/supervisor/courses/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      </div>
+
+      <div class="form-group">
+        <%= f.label :end_date %>
+        <%= f.date_field :end_date, class: 'datepicker', id: 'ed', readonly: true %>
+      </div>
+      
+      <div class="form-group">
+        <% Subject.all.each do |subject| -%>
+          <%= check_box_tag :subject_ids, subject.id, @course.subjects.include?(subject),
```
Comment (chuanhtuan) [inline app/views/supervisor/courses/new.html.erb]: @course.subjects.include?(subject)
I think, no need check include
please retry

Comment (zenith19) [inline app/views/supervisor/courses/new.html.erb]: I need to check include because otherwise all subject checkboxes are checked when the page load

Comment (chuanhtuan) [inline app/views/supervisor/courses/new.html.erb]: collection_check_boxes

**Pull request merged**